### PR TITLE
[helm-chart] sidecar container taraxa-indexer for rpc nodes

### PR DESCRIPTION
**Pull request by rjonczy**
## Purpose

Expose data from [taraxa-indexer](https://github.com/Taraxa-project/taraxa-indexer)

## How does the solution address the problem

Added [taraxa-indexer](https://github.com/Taraxa-project/taraxa-indexer) to RPC nodes and exposed via `Ingress`

## Changes made

Added:
- [taraxa-indexer](https://github.com/Taraxa-project/taraxa-indexer) as a sidecar container to RPC nodes
- new Ingress exposing Indexer's endpoints with sticky session based on cookie

### CODE REVIEW

File: charts/taraxa-node/templates/consensus-node.yaml [hunk 1]
```diff
@@ -44,8 +44,16 @@ spec:
               subPath: entrypoint.sh
         {{- end }}
         - name: config-adapter
+          {{- if and .Values.consensusnode.image.repository .Values.consensusnode.image.tag }}
+          image: "{{ .Values.consensusnode.image.repository }}:{{ .Values.consensusnode.image.tag }}"
+          {{- else }}
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          {{- end }}
+          {{- if .Values.consensusnode.image.imagePullPolicy }}
```
Comment (rattrap) [inline charts/taraxa-node/templates/consensus-node.yaml:52]: Wrong key. Should be `.Values.consensusnode.image.pullPolicy`

File: charts/taraxa-node/templates/consensus-node.yaml [hunk 2]
```diff
@@ -94,8 +102,16 @@ spec:
               subPath: status.py
         {{- end }}
         - name: consensus-node
+          {{- if and .Values.consensusnode.image.repository .Values.consensusnode.image.tag }}
+          image: "{{ .Values.consensusnode.image.repository }}:{{ .Values.consensusnode.image.tag }}"
+          {{- else }}
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          {{- end }}
+          {{- if .Values.consensusnode.image.imagePullPolicy }}
```
Comment (rattrap) [inline charts/taraxa-node/templates/consensus-node.yaml:110]: Wrong key. Should be `.Values.consensusnode.image.pullPolicy`

File: charts/taraxa-node/templates/taraxa-node.yaml [hunk 3]
```diff
@@ -35,8 +35,16 @@ spec:
     spec:
       initContainers:
         - name: config-adapter
+          {{- if and .Values.node.image.repository .Values.node.image.tag }}
+          image: "{{ .Values.node.image.repository }}:{{ .Values.node.image.tag }}"
+          {{- else }}
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          {{- end }}
+          {{- if .Values.node.image.imagePullPolicy }}
```
Comment (rattrap) [inline charts/taraxa-node/templates/taraxa-node.yaml:43]: Should be `.Values.node.image.pullPolicy`

File: charts/taraxa-node/templates/taraxa-node.yaml [hunk 4]
```diff
@@ -90,9 +98,31 @@ spec:
...
+            - name: indexer-data
+              mountPath: /data
+        {{- end }}
         - name: taraxa-node
+          {{- if and .Values.node.image.repository .Values.node.image.tag }}
+          image: "{{ .Values.node.image.repository }}:{{ .Values.node.image.tag }}"
+          {{- else }}
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          {{- end }}
+          {{- if .Values.node.image.imagePullPolicy }}
```
Comment (rattrap) [inline charts/taraxa-node/templates/taraxa-node.yaml:121]: Should be `.Values.node.image.pullPolicy`

**Review by rattrap — CHANGES_REQUESTED** [review]

File: charts/taraxa-node/values.yaml [hunk 5]
```diff
@@ -94,6 +95,11 @@ node:
     annotationsRpcWS: {}
     annotationsGraphQl: {}
     annotationsGraphQlWS: {}
+    annotationsIndexer:
+      nginx.ingress.kubernetes.io/affinity: "cookie"
+      nginx.ingress.kubernetes.io/session-cookie-name: "stickounet"
+      nginx.ingress.kubernetes.io/session-cookie-expires: "172800"
+      nginx.ingress.kubernetes.io/session-cookie-max-age: "172800"
```
Comment (rattrap) [inline charts/taraxa-node/values.yaml:102]: I think we should also add `nginx.ingress.kubernetes.io/affinity-mode: persistent` here

**Review by rattrap — CHANGES_REQUESTED** [review]

File: .circleci/config.yml [hunk 6]
```diff
@@ -236,6 +236,10 @@ commands:
                 --sort-by="~timestamp" \
                 --limit 1 \
                 --format="value(tags[0])" )
+              
+              # temp
+              PR_TAG=pr-2342-1a09cc9
+              # temp
```
[the remark this answers is not in the document]
Comment (rjonczy) [reply .circleci/config.yml]: removed. Will find another way for this.

Problem is that for helm related prs we are not building docker image of node itself. Will figure something out in separate pr.

File: .circleci/config.yml [hunk 7]
```diff
@@ -268,7 +272,12 @@ commands:
                 --set consensusnode.persistence.enabled=true \
                 --set node.serviceMonitor.enabled=true \
                 --set bootnode.serviceMonitor.enabled=false \
-                --set consensusnode.serviceMonitor.enabled=true
+                --set consensusnode.serviceMonitor.enabled=true \
+                --set node.nodeSelector."cloud\\.google\\.com/gke-nodepool"=blockchain-prnet \
+                --set bootnode.nodeSelector."cloud\\.google\\.com/gke-nodepool"=blockchain-prnet \
+                --set consensusnode.nodeSelector."cloud\\.google\\.com/gke-nodepool"=blockchain-prnet \
+                --set node.indexer.enabled=true \
+                --set node.indexer.image.tag=develop-503f613-1677511322
```
[the remark this answers is not in the document]
Comment (rjonczy) [reply .circleci/config.yml]: removed

File: charts/taraxa-node/templates/bootnode.yaml [hunk 8]
```diff
@@ -34,8 +34,17 @@ spec:
     spec:
       initContainers:
         - name: config-adapter
+          {{- if and .Values.bootnode.image.repository .Values.bootnode.image.tag }}
+          image: "{{ .Values.bootnode.image.repository }}:{{ .Values.bootnode.image.tag }}"
+          {{- else }}
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          {{- end }}
+
+          {{- if .Values.bootnode.image.imagePullPolicy }}
```
[the remark this answers is not in the document]
Comment (rjonczy) [reply charts/taraxa-node/templates/bootnode.yaml]: fixed

File: charts/taraxa-node/templates/consensus-node.yaml (hunk 1 above)
Comment (rjonczy) [reply charts/taraxa-node/templates/consensus-node.yaml]: fixed

File: charts/taraxa-node/templates/taraxa-node.yaml (hunk 3 above)
Comment (rjonczy) [reply charts/taraxa-node/templates/taraxa-node.yaml]: fixed

File: charts/taraxa-node/values.yaml (hunk 5 above)
Comment (rjonczy) [reply charts/taraxa-node/values.yaml:102]: added

File: .gitignore [hunk 9]
```diff
@@ -40,3 +40,7 @@ charts/*/charts/*.tgz
 
 # do not include generated documentation
 doxygen_docs
+
+# Helm stuff
+requirements.lock
+Chart.lock
```
[the remark this answers is not in the document]
Comment (rjonczy) [reply .gitignore]: moved.

**Review by rattrap — APPROVED** [review]

**Pull request merged**